Set-DomainUser.ps1

```diff
@@ -1,720 +1,1124 @@
+    [switch]$IgnoreShadowPermissionErrors,
+    [Parameter(
+        Mandatory = $false,
+        HelpMessage = "Name of Platform to be used for PSM accounts")]
+    [String]$PlatformName = "WIN-DOM-PSMADMIN-ACCOUNT",
+    [Parameter(
+        Mandatory = $false,
+        HelpMessage = "Name of Platform to be used for PSM accounts")]
+    [String]$PSMConnectAccountName = "PSMConnect",
+    [Parameter(
+        Mandatory = $false,
+        HelpMessage = "Name of Platform to be used for PSM accounts")]
+    [String]$PSMAdminConnectAccountName = "PSMAdminConnect",
+    [Parameter(
+        Mandatory = $false)]
+    [switch]$DoNotHarden,
+    [Parameter(
+        Mandatory = $false)]
+    [switch]$DoNotConfigureAppLocker
+)
+
+#Functions
+
+Function Get-DomainDnsName {
+    if ($env:USERDNSDOMAIN) {
+        return $env:USERDNSDOMAIN
+    }
+    else {
+        Write-Host "Unable to determine domain DNS name. Please provide it on the command line."
+        exit 1
+    }
+}
+
+Function Get-DomainNetbiosName {
+    if ($env:USERDOMAIN) {
+        return $env:USERDOMAIN
+    }
+    else {
+        Write-Host "Unable to determine domain NETBIOS name. Please provide it on the command line."
+        exit 1
+    }
+}
+
+Function Get-PvwaAddress {
+    <#
+    .SYNOPSIS
+    Backs up PSMConfig ps1 scripts
+    .DESCRIPTION
+    Copies PSM config items to -backup.ps1
+    .PARAMETER psmRootInstallLocation
+    PSM root installation folder
+    #>
+    param (
+        [Parameter(Mandatory = $true)]
+        $psmRootInstallLocation
+    )
+    try {
+        $VaultIni = Get-Content "$psmRootInstallLocation\vault\vault.ini"
+        $VaultIniAddressesLine = $VaultIni | Select-String "^Addresses"
+        $null = $VaultIniAddressesLine -match "(https://[0-9a-zA-Z][\.\-0-9a-zA-Z]*)"
+        $Address = $Matches[0]
+        If (!($Address)) {
+            Throw
+        }
+        return $Address
+    }
+    catch {
+        Write-Host "Unable to detect PVWA address automatically. Please rerun script and provide it using the -PvwaAddress parameter."
+        exit 1
+    }
+}
+
+
+Function ValidateCredentials {
+    <#
+    .SYNOPSIS
+    Tests whether the provided credentials are valid
+    .DESCRIPTION
+    Returns boolean test result, $true indicates success
+    .EXAMPLE
+    ValidateCredentials -Domain $domain -Credential $credential
+    #>
+    param(
+        [Parameter(Mandatory = $true)][string]$domain,
+        [Parameter(Mandatory = $true)][PSCredential]$Credential
+    )
+    Process {
+        try {
+            Add-Type -AssemblyName System.DirectoryServices.AccountManagement
+            $Directory = New-Object System.DirectoryServices.AccountManagement.PrincipalContext([System.DirectoryServices.AccountManagement.ContextType]::Domain, $domain)
+            return $Directory.ValidateCredentials($Credential.UserName, $Credential.GetNetworkCredential().Password)
+        }
+        catch {
+            return $false
+        }
+    }
+}
+
```

Comment:
Function to check whether the username and password provided for the PSM users is correct